### PR TITLE
Add Venting and Leak File inputs for subtypes

### LINKED ISSUE #43: Add ability to sample from leaks files with subtypes
Currently sampling from a leaks file can only be done when there is no subtypes file. Add the ability to specify the leaks files per subtype in the subtypes files

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
Added the option in the subtype file to provide venting and leak file inputs to sample from to inform vent and leak rates respectively.

Reason for change: Previously, sampling from leak rates was only possible at a program level. To increase simulation accuracy, it has been requested that the ability to sample at a subtype level should be added. For venting, sampling at a subtype level is also desired as an alternative to provided site level static venting rates.

Closes #43 

All E2E tests are passing: 
[testing_logs.zip](https://github.com/LDAR-Sim/LDAR_Sim/files/11752182/testing_logs.zip)

### CODE REVIEW

**Comment by ThomasGalesloot** [comment]
3 questions:
1. Should this be split into 2 PR's, one for leak files in the subtype file and 1 for vent files in the subtype file?
2. Should I upload another subtype_distributions file in the inputs folder to serve as an example?
3. As unit testing these code changes is difficult, should I add new end to end test(s) to test this functionality?

**Comment by SEAJang** [comment]
1. One PR is fine I think just because they build on the same feature
2. Another example that showcases the feature added is always useful. So yes.
3. Yes.

File: LDAR_Sim/src/utils/emis_inputs.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# You should have received a copy of the MIT License
+# along with this program.  If not, see <https://opensource.org/licenses/MIT>.
+#
+# ------------------------------------------------------------------------------
+import numpy as np
+import pandas as pd
+
+
+def assign_vents(program, wd) -> None:
+    for st_idx, subtype in program['subtypes'].items():
```
Comment (SEAJang) [inline LDAR_Sim/src/utils/emis_inputs.py:25]: Should there be doc string?

**Review by SEAJang — CHANGES_REQUESTED** [review]
Other than the minor changes related to documentation and the E2E testing mentioned in the PR comments, it looks good.

File: USER_MANUAL.md [hunk 2]
```diff
@@ -975,6 +975,129 @@ The NRd value should be the same for **all** programs
 
 **Notes of caution:** This functionality could be used to 'increase' the number of sites in a program, as sampling occurs with replacement. For example, if a program only had 5 sites, sampling of 100 sites would create the illusion of more sites in the program. However, the sites will only represent 5 locations, so if simulations include geographical considerations (e.g. road networks), this may not be advisable as travel times might not be meaningful.
 
+### subtype_file
+
+**Data type:** string
+
+**Default input:** None
+
+**Description:** Imports a lookup table that allows the modeler to set the leak rate source, vent_rate_file, NRd and LPR per facility according to subtype type. This functionality can be used when these values can be approximately estimated by site subtype. Each row represents a site subtype type and each column represents a value. If this parameter is set to None, or left blank, the described values are set at a program level. For more information on setting NRd and LPR see the entries in program level parameters for NRd and LPR.
```
[the remark this answers is not in the document]
Comment (ThomasGalesloot) [reply USER_MANUAL.md:984]: Modeler is actually used in the user manual - That is where I got it from

[the remark this answers is not in the document]
Comment (ThomasGalesloot) [reply USER_MANUAL.md:984]: Are we using the term "user" in other places?

File: LDAR_Sim/src/utils/emis_inputs.py (hunk 1 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/src/utils/emis_inputs.py:31]: Done

File: USER_MANUAL.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ThomasGalesloot) [reply USER_MANUAL.md:984]: Resolution: Both user and modeler are used. A separate change will be authored to make wording consistent.

File: changelog.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Change Log
 
+## 2023-06 - Version 2.1.3
+
+1. **Added ability to sample for leak rates, vent rates per subtype** Users can now specify vent rate and leak rate files to sample from on a per subtype basis in the subtype file.
+
```
[the remark this answers is not in the document]
Comment (ThomasGalesloot) [reply changelog.md:7]: Done

**Comment by ThomasGalesloot** [comment]
Sally's concerns/comments have now been addressed. Ready for review.

**Review by SEAJang — APPROVED** [review]
Looks good

**Review by KeeganShaw-GIS — APPROVED** [review]
Awesome work.

**Pull request merged**